### PR TITLE
Included python implementation of bellman ford algorithm

### PR DESCRIPTION
**Pull request by Odion-Sonny**
## Change Summary

Included python implementation of bellman ford algorithm

## Checklist

**If you haven't fulfilled the below requirements or even delete the entire checklist, your PR won't be reviewed and will be closed without notice.**

### General 

- [x] This Pull Request is all my own work. (You'll be blacklisted if you are caught for plagiarism.)
- [x] I've read [CONTRIBUTING.md](https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md)
- [x] I've applied LaTex for all variables, formulas and time / space complexity instead of using backticks
- [x] I've started the app locally and verified all the content and all links (if applicable) are accessible correctly
- [x] I've included Complexity Analysis (Time Complexity & Space Complexity).
- [x] I've written my explanation well and it is easy to understand for beginners

### Tutorial

- [x] I've read and followed the [Tutorial Template](https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md#tutorial-template)
- [x] I've explained my topic well with 2 - 3 LC problems and no external problems are used. 
- [x] I've provided the full working solutions to the problems used in this tutorial.
- [x] I've provided suggested problems at the end with the given format. See [here](https://raw.githubusercontent.com/wingkwong/leetcode-the-hard-way/main/tutorials/math/number-theory/binary-exponentiation.md) as an example. If the target solution is not available, leave `solutionLink` blank.
- [x] I've given credits / references if I use external resources. (For an image, give credit under it. Otherwise, add a new section called References at the end (after Suggested Problems).)

### Solutions

- [x] I've read and followed the [Solution Template](https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md#solution-template)
- [x] I've formatted my code well with [K&R Coding style](https://gist.github.com/jesseschalken/0f47a2b5a738ced9c845#why-kr)
- [x] I've confirmed that comments are put above each line rather than writing on the same line.
- [x] I've included a meaningful approach name for my solution. e.g. `## Approach 1: Two Pointers`.

### CODE REVIEW

**Review by wingkwong — CHANGES_REQUESTED** [review]
The Python code should be under a new TabItem. i.e.

````
<TabItem value="py" label="Python">
<SolutionAuthor name="@Odion-Sonny"/>

```py
def bellman_ford(g, dist, src, mx_edges):
    dist[src] = 0
    for i in range(mx_edges+1):
        ndist = dist.copy()
        for x in g:
            from_, to, cost = x
            ndist[to] = min(ndist[to], dist[from_] + cost)
        dist = ndist
```
</TabItem>
````

Please also verify locally first.

File: tutorials/graph-theory/bellman-ford-algorithm.md [hunk 1]
```diff
@@ -35,4 +35,20 @@ void bellman_ford(T_a3 &g, T_vector &dist, int src, int mx_edges) {
...
+<SolutionAuthor name="@Odion-Sonny"/>
+
+```py
+def bellman_ford(g, dist, src, mx_edges):
+    dist[src] = 0
+    for i in range(mx_edges+1):
+        ndist = dist.copy()
+        for x in g:
+            from_, to, cost = x
+            ndist[to] = min(ndist[to], dist[from_] + cost)
```
Comment (wingkwong) [inline tutorials/graph-theory/bellman-ford-algorithm.md:49]: prefer `_from`

File: tutorials/graph-theory/bellman-ford-algorithm.md [hunk 2]
```diff
@@ -35,4 +35,20 @@ void bellman_ford(T_a3 &g, T_vector &dist, int src, int mx_edges) {
...
-</Tabs>
\ No newline at end of file
+
+<TabItem value="py" label="Python">
+<SolutionAuthor name="@Odion-Sonny"/>
+
+```py
+def bellman_ford(g, dist, src, mx_edges):
+    dist[src] = 0
+    for i in range(mx_edges+1):
```
Comment (wingkwong) [inline tutorials/graph-theory/bellman-ford-algorithm.md:45]:
```
for i in range(mx_edges + 1):
```

File: tutorials/graph-theory/bellman-ford-algorithm.md [hunk 3]
```diff
@@ -35,4 +35,20 @@ void bellman_ford(T_a3 &g, T_vector &dist, int src, int mx_edges) {
...
 ```
 </TabItem>
-</Tabs>
\ No newline at end of file
+
+<TabItem value="py" label="Python">
+<SolutionAuthor name="@Odion-Sonny"/>
+
+```py
+def bellman_ford(g, dist, src, mx_edges):
```
Comment (wingkwong) [inline tutorials/graph-theory/bellman-ford-algorithm.md:43]: unlike c++, this function won't work as `dist` would stay in the scope of the function. You need to add the return statement.

```py
return dist
```

A simple verification:

```py
class Solution:
    def findCheapestPrice(self, n: int, flights: List[List[int]], src: int, dst: int, k: int) -> int:
        def bellman_ford(g, dist, src, mx_edges):
            dist[src] = 0
            for i in range(mx_edges + 1):
                ndist = dist.copy()
                for x in g:
                    _from, to, cost = x
                    ndist[to] = min(ndist[to], dist[_from] + cost)
                dist = ndist
            return dist
        g = []
        for f in flights: g.append([f[0], f[1], f[2]])
        cost = bellman_ford(g, [1e9] * n, src, k)
        if cost[dst] == 1e9: return -1
        return cost[dst]
```

**Review by wingkwong — CHANGES_REQUESTED** [review]

File: tutorials/graph-theory/bellman-ford-algorithm.md (hunk 3 above)
Comment (Odion-Sonny) [reply tutorials/graph-theory/bellman-ford-algorithm.md:43]: Thanks for pointing this out, that was a silly oversight.

File: tutorials/graph-theory/bellman-ford-algorithm.md (hunk 1 above)
Comment (Odion-Sonny) [reply tutorials/graph-theory/bellman-ford-algorithm.md]: Acknowledged

File: tutorials/graph-theory/bellman-ford-algorithm.md (hunk 2 above)
Comment (Odion-Sonny) [reply tutorials/graph-theory/bellman-ford-algorithm.md]: Acknowledged

**Pull request merged**